great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -85,9 +84,9 @@ def get(self, name: str) -> Checkpoint:
 
         return self._get(key=key)
 
-    def _get(self, key: GXCloudIdentifier | ConfigurationIdentifier) -> Checkpoint:
-        config: dict | CheckpointConfig = self._store.get(key=key)  # type: ignore[assignment]
-        if isinstance(config, CheckpointConfig):
-            config = config.to_json_dict()
+    def _get(self, key: GXCloudIdentifier | StringKey) -> Checkpoint:
+        checkpoint = self._store.get(key=key)
```

Comment:
Due to Pydantic model we're guaranteed to have `Checkpoint` but we need to do some type guarding to appease `mypy` (highlights a flaw in our store design).

Reply:
i see - well, looks like this approach makes testing more cumbersome, so I'll leave it up to you which approach you want to go with.